src/expr/src/scalar/func.rs

```diff
@@ -8102,6 +8190,9 @@ impl VariadicFunc {
                 ScalarType::Array(Box::new(ScalarType::String)).nullable(in_nullable)
             }
             RegexpReplace => ScalarType::String.nullable(in_nullable),
+            StringToArray => {
+                ScalarType::Array(Box::new(ScalarType::String)).nullable(input_types[0].nullable)
```

Comment:
do I need to protect against empty input_types here?

Reply:
No need, the length of `input_types` should correspond to the length of the argument list of a valid function call. You could add this fact as a doc comment on top of `output_type`.

Reply:
great, thanks!